x-pack/plugins/enterprise_search/server/routes/app_search/role_mappings.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import { RouteDependencies } from '../../plugin';
+
+const roleMappingBaseSchema = {
+  rules: schema.recordOf(schema.string(), schema.string()),
+  roleType: schema.string(),
+  engines: schema.arrayOf(schema.string()),
+  accessAllEngines: schema.boolean(),
+  authProvider: schema.arrayOf(schema.string()),
+};
+
+export function registerRoleMappingsRoute({
+  router,
+  enterpriseSearchRequestHandler,
+}: RouteDependencies) {
+  router.get(
+    {
+      path: '/api/app_search/role_mappings',
+      validate: false,
+    },
+    enterpriseSearchRequestHandler.createRequest({
+      path: '/role_mappings',
+    })
+  );
+
+  router.post(
+    {
+      path: '/api/app_search/role_mappings',
+      validate: {
+        body: schema.object(roleMappingBaseSchema),
+      },
+    },
+    enterpriseSearchRequestHandler.createRequest({
+      path: '/role_mappings',
+    })
+  );
+}
+
+export function registerRoleMappingRoute({
+  router,
+  enterpriseSearchRequestHandler,
+}: RouteDependencies) {
+  router.get(
+    {
+      path: '/api/app_search/role_mappings/{id}',
+      validate: {
+        params: schema.object({
+          id: schema.string(),
+        }),
+      },
+    },
+    enterpriseSearchRequestHandler.createRequest({
+      path: '/role_mappings/:id',
+    })
+  );
+
+  router.put(
+    {
+      path: '/api/app_search/role_mappings/{id}',
+      validate: {
+        body: schema.object({
+          ...roleMappingBaseSchema,
+          id: schema.string(),
+        }),
+        params: schema.object({
+          id: schema.string(),
+        }),
+      },
+    },
+    enterpriseSearchRequestHandler.createRequest({
+      path: '/role_mappings/:id',
+    })
+  );
+
+  router.delete(
+    {
+      path: '/api/app_search/role_mappings/{id}',
+      validate: {
+        params: schema.object({
+          id: schema.string(),
+        }),
+      },
+    },
+    enterpriseSearchRequestHandler.createRequest({
+      path: '/role_mappings/:id',
+    })
+  );
+}
+
+export function registerNewRoleMappingRoute({
+  router,
+  enterpriseSearchRequestHandler,
+}: RouteDependencies) {
+  router.get(
+    {
+      path: '/api/app_search/role_mappings/new',
+      validate: false,
+    },
+    enterpriseSearchRequestHandler.createRequest({
+      path: '/role_mappings/new',
+    })
+  );
+}
+
+export function registerResetRoleMappingRoute({
```

Comment:
`registerRoleMappingsRoutes` is at the bottom of the file. These are merely exported so they can be tested individually

Reply:
Glad you pointed this out. I cherry-picked my main branch and forgot to get the main index file changes. Perhaps this will address you comment above. Nice catch! (https://github.com/elastic/kibana/pull/94221/commits/2f8e242bd1b574dd4656ed18d1ebc3ad0785334f)